openmc/deplete/abc.py

```diff
@@ -638,27 +650,52 @@ def __init__(self, operator, timesteps, power=None, power_density=None):
                         self._num_stages))
         self.operator = operator
         self.chain = operator.chain
-        if not isinstance(timesteps, Iterable):
-            self.timesteps = [timesteps]
-        else:
-            self.timesteps = timesteps
+
+        # Determine power and normalize units to W
         if power is None:
             if power_density is None:
                 raise ValueError("Either power or power density must be set")
             if not isinstance(power_density, Iterable):
                 power = power_density * operator.heavy_metal
             else:
-                power = [p * operator.heavy_metal for p in power_density]
-
+                power = [p*operator.heavy_metal for p in power_density]
         if not isinstance(power, Iterable):
             # Ensure that power is single value if that is the case
-            power = [power] * len(self.timesteps)
-        elif len(power) != len(self.timesteps):
+            power = [power] * len(timesteps)
+
+        if len(power) != len(timesteps):
             raise ValueError(
                 "Number of time steps != number of powers. {} vs {}".format(
-                    len(self.timesteps), len(power)))
+                    len(timesteps), len(power)))
 
-        self.power = power
+        # Get list of times / units
+        if isinstance(timesteps[0], Iterable):
+            times, units = zip(*timesteps)
+        else:
+            times = timesteps
+            units = [timestep_units] * len(timesteps)
+
+        # Determine number of seconds for each timestep
+        seconds = []
+        for time, unit, watts in zip(times, units, power):
```

Comment:
I would recommend a check for non-negative reals here. Passing a string goes allowed until `asarray` (unless all are strings (in which case, :scream:)). It _should_ go without saying that power is non-negative, but a light check here would prevent serious errors